### ISSUE TITLE
Prove STRIP for autotools

### ISSUE THREAD

**Issue by RReverser** [emscripten-core/emscripten#20307]
Autotools use `STRIP` variable (https://github.com/autotools-mirror/automake/blob/80714ea0aba62f025780f432abfbab2e66f6f801/m4/strip.m4#L25) which allows to correctly strip binaries when cross-compiling instead of attempting to run regular `strip` on incompatible binaries.

I thought I could relatively easily add this here https://github.com/emscripten-core/emscripten/blob/7d7dbdef08c6a418d5256b89edc2b7be030b0b80/tools/building.py#L70, pointing to `llvm-strip` or `emstrip`, but the problem is that it will be invoked by autotools on executables themselves, which in our case will usually happen to be a JS file.

I wonder if it would make sense to modify `emstrip` (https://github.com/emscripten-core/emscripten/blob/0c21eb984593d4bf606f0e7fa6c396059ccf4860/emstrip.py#L7) to automatically find `.wasm` file if the provided one looks like `.js`, so that it correctly handles both standalone Wasm and JS+Wasm cases? Then we would be able to provide such `emstrip` wrapper as `STRIP` executable.

cc @sbc100

**Comment by sbc100** [comment]
Sounds like a great idea!

**Comment by RReverser** [comment]
> to automatically find `.wasm` file if the provided one looks like `.js`, so that it correctly handles both standalone Wasm and JS+Wasm cases

Is there any prior work I could reuse for this? I imagine `NM` in the linked code could also benefit from such search, but for now it just uses `llvm-nm`.

**Comment by RReverser** [comment]
(alternatively, is this something you want to take on?)

**Comment by sbc100** [comment]
Not that I know of.   I imagine it would just be a case of modifying `emstrip.py` such that it inspects all its non-argument inputs.   If it finds one that looks like a .js file, and there is a corresponding `.wasm` next to it then it would simply replace the sys.argv element before calling the underlying llvm-strip?

**Comment by RReverser** [comment]
Yeah that sounds about right... I'll probably keep this open for now as it's low-priority (from what I can see, most users just disable stripping for now) but might try later if no one else takes this.

**Comment by RReverser** [comment]
> If it finds one that looks like a .js file, and there is a corresponding `.wasm` next to it

Just adding for the future implementer: it's probably difficult to determine what looks like a `.js` file, so instead it might be better to try and check Wasm magic signature at the beginning of the file - if it's not there, it's likely `.js` and extra search is needed, otherwise use as-is.
